public/src/overmind/actions.tsx

```diff
@@ -913,3 +795,20 @@ export const updateGroupName = ({ state }: Context, name: string): void => {
 export const setConnectionStatus = ({ state }: Context, status: ConnStatus) => {
     state.connectionStatus = status
 }
+
+// setSubmissionOwner sets the owner of the currently selected submission.
+// The owner is either an enrollment or a group.
+export const setSubmissionOwner = ({ state }: Context, { submission, enrollment }: { submission: Submission, enrollment: Enrollment | Group }) => {
+    if (submission.groupID > 0) {
+        // If the submission has a groupID, it is a group submission,
+        // regardless of whether the enrollment object is of group or enrollment type.
+        state.submissionOwner = { type: "GROUP", id: submission.groupID }
+    } else if (enrollment instanceof Group) {
+        // If the enrollment object is a group, it is a group submission.
+        // However, the submission does not have a groupID.
+        // This should not happen, but we handle it anyway.
+        return
+    } else {
+        state.submissionOwner = { type: "ENROLLMENT", id: enrollment.ID }
```

Comment:
Doesn't look like this submission type is "checked against". That's okay.

My actual question is: Should this type be "USER" instead of "ENROLLMENT" since the group type is handled separately? Or can this also be a group enrollment?

Reply:
We can rename the type `"ENROLLMENT"` to `"USER"`. The `type` determines which map to look for submissions in. If the `enrollment` is of the type `Enrollment`, we look for the `id` in `userSubmissions: Map<bigint, Submissions>`. If we're dealing with a `Group`, `groupSubmissions: Map<bigint, Submissions>` is used instead.